src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,28 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  var foundKey;
+  for (let key of Object.keys(config.carry.sizes).sort(function(a, b) { return parseInt(a, 10) - parseInt(b, 10); })) {
```

Comment:
Well it is an object not an array AFAIK they are not sorted, so here it gets a bit more tricky. I think we can leave it like this right now. When checking the performance next time either it pops up again or it is not so bad.